tests/drivers/gpio/egpio_more_loops/src/main.c

```diff
@@ -0,0 +1,746 @@
+	zassert_equal(rc, 0, "gpio_pin_set(dev, PIN_OUT_1, %d) failed", value & BIT(0));
+
+	/* OUT_2 */
+	rc = gpio_pin_set(dev_out, PIN_OUT_2, value & BIT(1));
+	zassert_equal(rc, 0, "gpio_pin_set(dev, PIN_OUT_2, %d) failed", value & BIT(1));
+
+	/* OUT_3 */
+	rc = gpio_pin_set(dev_out, PIN_OUT_3, value & BIT(2));
+	zassert_equal(rc, 0, "gpio_pin_set(dev, PIN_OUT_3, %d) failed", value & BIT(2));
+
+	/* OUT_4 */
+	rc = gpio_pin_set(dev_out, PIN_OUT_4, value & BIT(3));
+	zassert_equal(rc, 0, "gpio_pin_set(dev, PIN_OUT_4, %d) failed", value & BIT(3));
+
+	/* OUT_5 */
+	rc = gpio_pin_set(dev_out, PIN_OUT_5, value & BIT(4));
+	zassert_equal(rc, 0, "gpio_pin_set(dev, PIN_OUT_5, %d) failed", value & BIT(4));
+
+	/* Check inputs */
+	_check_inputs(value);
+}
+
+/* Set output GPIOs with gpio_pin_toggle()
+ *and confirm that input GPIOs have expected state.
+ */
+static void _gpio_pin_toggle_and_check(uint32_t value)
+{
+	int rc;
+	gpio_port_pins_t port_pins;
+
+	/* Set initial state 00000 */
+	port_pins = BIT(PIN_OUT_5) | BIT(PIN_OUT_4) | BIT(PIN_OUT_3)
+				| BIT(PIN_OUT_2) | BIT(PIN_OUT_1);
+	rc = gpio_port_set_masked_raw(dev_out, port_pins, 0);
+	zassert_equal(rc, 0, "gpio_port_set_masked_raw(port, %d, %d) failed",
+		port_pins, 0);
+
+	/* OUT_1 */
+	if (value & BIT(0)) {
+		rc = gpio_pin_toggle(dev_out, PIN_OUT_1);
+		zassert_equal(rc, 0, "gpio_pin_toggle(dev_out, PIN_OUT_1) failed");
+	}
+
+	/* OUT_2 */
+	if (value & BIT(1)) {
+		rc = gpio_pin_toggle(dev_out, PIN_OUT_2);
+		zassert_equal(rc, 0, "gpio_pin_toggle(dev_out, PIN_OUT_2) failed");
+	}
+	/* OUT_3 */
+	if (value & BIT(2)) {
+		rc = gpio_pin_toggle(dev_out, PIN_OUT_3);
+		zassert_equal(rc, 0, "gpio_pin_toggle(dev_out, PIN_OUT_3) failed");
+	}
+	/* OUT_4 */
+	if (value & BIT(3)) {
+		rc = gpio_pin_toggle(dev_out, PIN_OUT_4);
+		zassert_equal(rc, 0, "gpio_pin_toggle(dev_out, PIN_OUT_4) failed");
+	}
+	/* OUT_5 */
+	if (value & BIT(4)) {
+		rc = gpio_pin_toggle(dev_out, PIN_OUT_5);
+		zassert_equal(rc, 0, "gpio_pin_toggle(dev_out, PIN_OUT_5) failed");
+	}
+
+	/* Check inputs */
+	_check_inputs(value);
+}
+
+/* Generate "mooving 1" pattern on outputs.
+ * Set outputs (and check inputs) using function provided in the fn parameter.
+ */
+static void _generic_mooving_1(void (*fn)(uint32_t))
+{
+	/* Set/check 11111 */
+	(*fn)(0x11U);
+
+	/* Set/check 00000 */
+	(*fn)(0x0U);
+
+	/* Set/check 00001 */
+	(*fn)(0x1U);
+
+	/* Set/check 00010 */
+	(*fn)(0x2U);
+
+	/* Set/check 00100 */
+	(*fn)(0x4U);
+
+	/* Set/check 01000 */
+	(*fn)(0x8U);
+
+	/* Set/check 10000 */
+	(*fn)(0x10U);
+
+	/* Set/check 01000 */
+	(*fn)(0x8U);
+
+	/* Set/check 00100 */
+	(*fn)(0x4U);
+
+	/* Set/check 00010 */
+	(*fn)(0x2U);
+
+	/* Set/check 00001 */
+	(*fn)(0x1U);
+
+	/* Set/check 00000 */
+	(*fn)(0x0U);
+}
+
+/**
+ * @brief test setting output GPIO high/low with
+ * gpio_pin_configure()
+ */
+ZTEST(egpio_extended, test_mooving_1_with_configure_output)
+{
+	_generic_mooving_1(&_gpio_pin_configure_and_check);
+}
```

Comment:
I think it would be much more readable when just iterating over an array for each function.

Reply:
My idea was to have separate test for each GPIO API function.
In case error is specific to just one of them, remaining tests will pass.
Otherwise, there is just single result pass/fail from all calls.